test/verify_extra_test.c

```diff
@@ -321,6 +339,8 @@ int setup_tests(void)
     ADD_TEST(test_self_signed_error);
     ADD_TEST(test_purpose_ssl_client);
     ADD_TEST(test_purpose_ssl_server);
+    ADD_TEST(test_purpose_ssl_client_override_fail);
+    ADD_TEST(test_purpose_ssl_client_override_pass);
```

Comment:
I miss tests for `X509_STORE_CTX_set_trust_ex()`

Reply:
Normally, for such rather simple functions thorough tests would not be required,
but given the critical mistake so far in this PR which even got an approval,
you should somehow demonstrate that you know well what you are doing.
Yet this does not strictly need to be done via deep testing - you can also convince me otherwise.